### ISSUE TITLE
[cloud_firestore] `Query.orderBy(FieldPath.documentId)` throws exception at v0.14.0

### ISSUE THREAD

**Issue by mono0926** [FirebaseExtended/flutterfire#3210]
**Describe the bug**

`Query.orderBy(FieldPath.documentId)` stated throwing exception at v0.14.0.
It works well until v0.13.7+1.

**To Reproduce**
1. Run this.
```dart
FirebaseFirestore.instance
    .collection('foo')
    .orderBy(FieldPath.documentId)
    .get();
```
1. Error log
```
E/flutter (12130): [ERROR:flutter/lib/ui/ui_dart_state.cc(171)] Unhandled Exception: type 'FieldPathType' is not a subtype of type 'FieldPath'
E/flutter (12130): #0      Query.orderBy (package:cloud_firestore/src/query.dart:238:15)
E/flutter (12130): #1      _DropdownButton.build.<anonymous closure> (package:example/pages/paging_page/paging_page.dart:119:14)
E/flutter (12130): #2      PopupMenuButtonState.showButtonMenu.<anonymous closure> (package:flutter/src/material/popup_menu.dart:1120:28)
E/flutter (12130): #3      _rootRunUnary (dart:async/zone.dart:1198:47)
E/flutter (12130): #4      _CustomZone.runUnary (dart:async/zone.dart:1100:19)
E/flutter (12130): #5      _FutureListener.handleValue (dart:async/future_impl.dart:143:18)
E/flutter (12130): #6      Future._propagateToListeners.handleValueCallback (dart:async/future_impl.dart:696:45)
E/flutter (12130): #7      Future._propagateToListeners (dart:async/future_impl.dart:725:32)
E/flutter (12130): #8      Future._completeWithValue (dart:async/future_impl.dart:529:5)
E/flutter (12130): #9      Future._asyncCompleteWithValue.<anonymous closure> (dart:async/future_impl.dart:567:7)
E/flutter (12130): #10     _rootRun (dart:async/zone.dart:1190:13)
E/flutter (12130): #11     _CustomZone.run (dart:async/zone.dart:1093:19)
E/flutter (12130): #12     _CustomZone.runGuarded (dart:async/zone.dart:997:7)
E/flutter (12130): #13     _CustomZone.bindCallbackGuarded.<anonymous closure> (dart:async/zone.dart:1037:23)
E/flutter (12130): #14     _microtaskLoop (dart:async/schedule_microtask.dart:41:21)
E/flutter (12130): #15     _startMicrotaskLoop (dart:async/schedule_microtask.dart:50:5)
```

**Expected behavior**

There are no errors.

**Additional context**

There's a problem here.

`field_path.dart`

```dart
  static FieldPathType get documentId {
    return FieldPathType.documentId;
  }
```

`FieldPath.documentId` is `FieldPathType`, but it is tried to be assigned to `FieldPath`.

`query.dart`

```dart
  Query orderBy(dynamic field, {bool descending = false}) {
    assert(field != null && descending != null);
    _assertValidFieldType(field);
    assert(!_hasStartCursor(),
        "Invalid query. You must not call startAt(), startAtDocument(), startAfter() or startAfterDocument() before calling orderBy()");
    assert(!_hasEndCursor(),
        "Invalid query. You must not call endAt(), endAtDocument(), endBefore() or endBeforeDocument() before calling orderBy()");

    final List<List<dynamic>> orders =
        List<List<dynamic>>.from(parameters['orderBy']);

    assert(orders.where((List<dynamic> item) => field == item[0]).isEmpty,
        "OrderBy field '$field' already exists in this query");

   // Type mismatch here (FieldPath vs FieldPathType)
    FieldPath fieldPath = field is String ? FieldPath.fromString(field) : field;
```

**Flutter doctor**
Run `flutter doctor` and paste the output below:

```
❯ flutter doctor
Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel dev, 1.21.0-9.0.pre, on Mac OS X 10.15.6 19G73, locale en-JP)
 
[✓] Android toolchain - develop for Android devices (Android SDK version 29.0.3)
[✓] Xcode - develop for iOS and macOS (Xcode 11.6)
[✓] Chrome - develop for the web
[✓] Android Studio (version 4.0)
[✓] IntelliJ IDEA Community Edition (version 2019.1.3)
[✓] VS Code (version 1.47.3)
[✓] Connected device (6 available)
    ! Error: monoPhone is not connected. Xcode will continue when monoPhone is connected. (code -13)
    ! Error: monoyagon is not connected. Xcode will continue when monoyagon is connected. (code -13)
    ! Error: monoPhone7Plus has recently restarted. Xcode will continue when monoPhone7Plus is unlocked. (code -14)

• No issues found!
```

**Comment by mono0926** [comment]
@Ehesp 

A `FieldPath.documentId` related error was fixed partically at #3200, but `.orderBy(FieldPath.documentId)` is still broken.

**Comment by Salakar** [comment]
Release with a fix is now published. Latest versions can be seen [here](https://firebase.flutter.dev/docs/migration/#2-update-firebase-plugins). 

Thanks again for the bug report :)
